### PR TITLE
Packed batch<=>space ND: transpose, slice, pad, 5-D and 6-D packed sampling.

### PR DESCRIPTION
**Pull request by astojilj**
Packed transpose, slice, pad and 5-D and 6-D packed sampling, in this PR, implement support for packed batchToSpaceND and packed spaceToBatchND.
It is verified using DeepLabV3+ MobileNetV2 segmentation, with exclusion of packed batch normalization (erroneous).

Instead of adding additional tests, existing transpose, slice, pad, batchToSpaceND and spaceToBatchND tests are reused for packed operations using activateWebGLPackedTestEnv. activateWebGLPackedTestEnv and deactivateWebGLPackedTestEnv delimit set of tests to be run in both packed and non-packed environment mode. When majority of operations get their packed variants, this could be replaced by running additional packed env for all tests.

Didn't define separate WEBGL_PACK_XXXX for transpose, slice and pad - using master switch WEBGL_PACK for all of them.

PERF

---

##### For repository owners only:

Please remember to apply all applicable tags to your pull request.
Tags: FEATURE, BREAKING, BUG, PERF, DEV, DOC, SECURITY

For more info see: https://github.com/tensorflow/tfjs/blob/master/DEVELOPMENT.md

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/1448)

### CODE REVIEW

**Comment by annxingyuan** [comment]
Hey @astojilj - this is amazing! We're super excited to get this contribution into the library, but would you mind splitting it up? This will make it easier to review and easier to bisect for future debugging. Something like:

- batchToSpace
- transpose
- slice
- pad

You could introduce the infrastructural changes (including packed 5D / 6D sampling) along with one of the ops, and I'll make sure to review that one first.

**Comment by astojilj** [comment]
@annxingyuan,
I'm back from vacation and rechecking this - should I wait for [WIP Packed test env #1469](https://github.com/tensorflow/tfjs-core/pull/1469) before splitting this?

**Comment by astojilj** [comment]
> @astojilj Can we close this PR since you've split it up?

@annxingyuan, certainly. Thanks.
